File: code_samples/elixir/data_structures_hash_table.ex
# Data Structures: Hash Table
# AI/ML Training Sample

defmodule HashTable do
  defstruct data: ""
  
  def new(), do: %HashTable{}
  
  def process(%HashTable{} = struct, input) do
    %{struct | data: input}
  end
  
  def get_data(%HashTable{data: data}), do: data
  
  def validate(%HashTable{data: data}) do
    String.length(data) > 0
  end
end

# Example usage
instance = HashTable.new()
updated = HashTable.process(instance, "example")
IO.puts("Data: " <> HashTable.get_data(updated))
IO.puts("Valid: " <> to_string(HashTable.validate(updated)))

File: code_samples/java/DataStructuresHashTable.java
/**
 * Data Structures: Hash Table
 * AI/ML Training Sample
 */
public class DataStructuresHashTable {
    
    private String data;
    
    public DataStructuresHashTable() {
        this.data = "";
    }
    
    public void process(String input) {
        this.data = input;
    }
    
    public String getData() {
        return this.data;
    }
    
    public boolean validate() {
        return this.data != null && !this.data.isEmpty();
    }
    
    public static void main(String[] args) {
        DataStructuresHashTable instance = new DataStructuresHashTable();
        instance.process("example");
        System.out.println("Data: " + instance.getData());
        System.out.println("Valid: " + instance.validate());
    }
}
